hie-plugin-api/Haskell/Ide/Engine/PluginsIdeMonads.hs

```diff
@@ -451,6 +454,17 @@ withIndefiniteProgress t c f = do
   lf <- ideEnvLspFuncs <$> getIdeEnv
   control $ \run -> Core.withIndefiniteProgress lf t c (run f)
 
+-- | Sends a telemetry/event notification to the client, if the environment variable
+-- @HIE_TELEMETRY@ is set.
```

Comment:
Do we document all these various environment variables we use to configure the tests anywhere?

It might be really good to dump them via a CLI flag, but perhaps we should consider some mechanism that gathers them all in one place, so we can be sure that we have a single list. Some kind of structure with a list of used environment variables, and an API to request it, which fails if the variable is not registered.